ubidots/ubidots_in.js

```diff
@@ -1,94 +1,200 @@
-module.exports = function (RED) {
-  var mqtt = require('mqtt')
-
-  function getClient (self, endpointUrl, labelDevice, labelVariable, token) {
-    self.status({ fill: 'green', shape: 'ring', text: 'ubidots.connecting' })
-
-    var client = mqtt.connect('mqtt://' + endpointUrl, { username: token, password: '' })
-
-    client.on('error', function () {
-      client.end(true, function () {})
-      self.status({ fill: 'red', shape: 'ring', text: 'ubidots.error_connecting' })
-    })
-
-    client.on('close', function () {
-      client.end(true, function () {})
-    })
-
-    client.on('reconnect', function () {
-      var topic = '/v1.6/devices/' + labelDevice + '/' + labelVariable
-      var options = {}
-
-      self.status({ fill: 'green', shape: 'dot', text: 'ubidots.connected' })
-      options[topic] = 1
-
-      client.subscribe(options, function () {
+module.exports = function(RED) {
+  var mqtt = require("mqtt");
+  var fs = require("fs");
+  var path = require("path");
+
+  function getClient(
+    self,
+    topics,
+    useTLS,
+    endpointUrl,
+    labelDevice,
+    labelVariable,
+    token
+  ) {
+    self.status({ fill: "green", shape: "ring", text: "ubidots.connecting" });
+    var URL_PREFIX = "mqtt://";
+    var port = 1883;
+    var portTLS = 8883;
+    var certificate = fs.readFileSync(
+      path.join(__dirname, "./certificate.pem"),
```

Comment:
Why don't you use the certificate in the `keys` folder? This way we could prevent duplicating the file.